transactions/account_linking/add_as_child_from_claimed_auth_account_cap.cdc

```diff
@@ -1,78 +1,96 @@
+        handlerPathSuffix: String
     ) {
 
-    let managerRef: &ChildAccount.ChildAccountManager
-    let info: ChildAccount.ChildAccountInfo
-    let childAccountCap: Capability<&AuthAccount>
+    let collectionRef: &LinkedAccounts.Collection
+    let info: LinkedAccountMetadataViews.AccountInfo
+    let authAccountCap: Capability<&AuthAccount>
 
     prepare(signer: AuthAccount) {
-        // Get ChildAccountManager Capability, linking if necessary
-        if signer.borrow<
-                &ChildAccount.ChildAccountManager
-            >(
-                from: ChildAccount.ChildAccountManagerStoragePath
-            ) == nil {
-            // Save a ChildAccountManager to the signer's account
-            signer.save(<-ChildAccount.createChildAccountManager(), to: ChildAccount.ChildAccountManagerStoragePath)
+        /** --- Configure Collection & get ref --- */
+        //
+        // Check that Collection is saved in storage
+        if signer.type(at: LinkedAccounts.CollectionStoragePath) == nil {
+            signer.save(
+                <-LinkedAccounts.createEmptyCollection(),
+                to: LinkedAccounts.CollectionStoragePath
+            )
         }
-        // Ensure ChildAccountManagerViewer is linked properly
+        // Link the public Capability
         if !signer.getCapability<
-                &{ChildAccount.ChildAccountManagerViewer}
-            >(
-                ChildAccount.ChildAccountManagerPublicPath
-            ).check() {
-            // Link
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, MetadataViews.ResolverCollection}
+            >(LinkedAccounts.CollectionPublicPath).check() {
+            signer.unlink(LinkedAccounts.CollectionPublicPath)
+            signer.link<&LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, MetadataViews.ResolverCollection}>(
+                LinkedAccounts.CollectionPublicPath,
+                target: LinkedAccounts.CollectionStoragePath
+            )
+        }
+        // Link the private Capability
+        if !signer.getCapability<
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, NonFungibleToken.CollectionPublic, NonFungibleToken.Receiver, NonFungibleToken.Provider, MetadataViews.ResolverCollection}
+            >(LinkedAccounts.CollectionPrivatePath).check() {
+            signer.unlink(LinkedAccounts.CollectionPrivatePath)
             signer.link<
-                &{ChildAccount.ChildAccountManagerViewer}
+                &LinkedAccounts.Collection{LinkedAccounts.CollectionPublic, NonFungibleToken.CollectionPublic, NonFungibleToken.Receiver, NonFungibleToken.Provider, MetadataViews.ResolverCollection}
             >(
-                ChildAccount.ChildAccountManagerPublicPath,
-                target: ChildAccount.ChildAccountManagerStoragePath
+                LinkedAccounts.CollectionPrivatePath,
+                target: LinkedAccounts.CollectionStoragePath
             )
         }
```

Comment:
Are you actually using the private capability anywhere in this transaction?

Reply:
I'm not, but since this might be the first time a user links an account, I figured it would be best to go the full length of configuring the Collection resource instead of requiring two separate transactions - setup then link